File: linegraph/src/main/java/ir/farshid_roohi/utilites/GraphPath.java
package ir.farshid_roohi.utilites;

import android.graphics.Path;
import android.graphics.PointF;

public class GraphPath extends Path {
	
	private MatrixTranslator mMt;
	
	public GraphPath(int width, int height, int paddingLeft, int paddingBottom) {
		mMt = new MatrixTranslator(width, height, paddingLeft, paddingBottom);
	}
	
	@Override
	public void moveTo(float x, float y) {
		super.moveTo(mMt.calcX(x), mMt.calcY(y));
	}
	
	public void moveTo(PointF point) {
		super.moveTo(mMt.calcX(point.x), mMt.calcY(point.y));
	}
	
	@Override
	public void lineTo(float x, float y) {
		super.lineTo(mMt.calcX(x), mMt.calcY(y));
	}
	
	public void lineTo(PointF point) {
		super.lineTo(mMt.calcX(point.x), mMt.calcY(point.y));
	}
}

File: linegraph/src/main/java/io/github/utilites/GraphPath.kt
package io.github.utilites

import android.graphics.Path
import android.graphics.PointF

class GraphPath(width: Int, height: Int, paddingLeft: Int, paddingBottom: Int) : Path() {

    private val mMt: MatrixTranslator = MatrixTranslator(width, height, paddingLeft, paddingBottom)

    override fun moveTo(x: Float, y: Float) {
        super.moveTo(mMt.calcX(x), mMt.calcY(y))
    }

    fun moveTo(point: PointF) {
        super.moveTo(mMt.calcX(point.x), mMt.calcY(point.y))
    }

    override fun lineTo(x: Float, y: Float) {
        super.lineTo(mMt.calcX(x), mMt.calcY(y))
    }

    fun lineTo(point: PointF) {
        super.lineTo(mMt.calcX(point.x), mMt.calcY(point.y))
    }
}
